### PR TITLE
feat: add `isAccessibilityServiceEnabled`

### LINKED ISSUE #30863: Android: System Property Detection Detect State change
## Requires API Proposal
This issue may require a new API, or a change to an existing API. An API proposal should be added and discussed before proceeding with implementation. The API proposal can be added in the comments of this issue or linked as a separate issue.

## Description
Currently the AccessibilityInfo API can only detect "screenReaderChanged", and not other services. It would be nice to be able to detect other services such as Select to Speak, Switch Access, Voice Access, etc.

## React Native version:
v0.63

## Expected Behavior
The ability to detect a general change in the accessibility state of the system should be supported in the AccessibilityInfo API.

## Android Details
The AccessibilityManager class has a general "accessibilityStateChange" listener, that is triggered when any accessibility service changes from active to inactive or vice-versa.
https://developer.android.com/reference/android/view/accessibility/AccessibilityManager#addAccessibilityStateChangeListener(android.view.accessibility.AccessibilityManager.AccessibilityStateChangeListener)

Note: The currently supported "screenReaderChanged" method maps to the TouchExplorationStateChangeListener.

### PR DESCRIPTION
**Pull request by grgr-dkrk**
## Summary

fix #30863

This PR adds `isAccessibilityServiceEnabled` to get if accessibility services are enabled on Android.

## Changelog

[Android] [Added] - Added `isAccessibilityServiceEnabled` to get if accessibility services are enabled

## Test Plan

accessibilityService

### CODE REVIEW

**Comment by amarlette** [comment]
Thank you for submitting this @grgr-dkrk! I've added your PR to the Improved React Native Accessibility board. Thank you for your contribution to a more accessible React Native.

File: Libraries/Components/AccessibilityInfo/AccessibilityInfo.js [hunk 1]
```diff
@@ -33,6 +33,7 @@ type AccessibilityEventDefinitions = {
   change: [boolean], // screenReaderChanged
   reduceMotionChanged: [boolean],
   screenReaderChanged: [boolean],
+  accessibilityServiceChanged: [boolean],
```
Comment (kacieb) [inline Libraries/Components/AccessibilityInfo/AccessibilityInfo.js:36]: If this event is only supported on Android, we should mark that (perhaps by adding a new type `AccessibilityEventDefinitionsAndroid` like iOS has.

**Comment by kacieb** [comment]
This is interesting - so this listens for any change? cc @blavalla I know we discussed this a bit, would the potential use case for this be to listen for changes so as to be able to update the layout (or similar) if, for example, TalkBack was turned on/off? So you'd need to listen for changes, and then check if TalkBack is on with another function call.

**Comment by blavalla** [comment]
> This is interesting - so this listens for any change? cc @blavalla I know we discussed this a bit, would the potential use case for this be to listen for changes so as to be able to update the layout (or similar) if, for example, TalkBack was turned on/off? So you'd need to listen for changes, and then check if TalkBack is on with another function call.

Yeah, thats the general idea. Although we already have a method specifically for talkback with `screenReaderChanged`.  This would be able to detect *any* service changing, since on Android there can be 3rd party accessibility services, the API google provides doesn't have methods for specific services (talkback, switch access, etc.) like iOS does.

**Pull request closed without merging**